phpBB/includes/functions_user.php

```diff
@@ -968,482 +964,30 @@ function user_ban($mode, $ban, $ban_len, $ban_len_other, $ban_exclude, $ban_reas
-			$db->sql_query($sql);
-		}
-
-		unset($banlist_ary_tmp);
-	}
-	$db->sql_freeresult($result);
-
-	// We have some entities to ban
-	if (count($banlist_ary))
-	{
-		$sql_ary = array();
-
-		foreach ($banlist_ary as $ban_entry)
-		{
-			$sql_ary[] = array(
-				$type				=> $ban_entry,
-				'ban_start'			=> (int) $current_time,
-				'ban_end'			=> (int) $ban_end,
-				'ban_exclude'		=> (int) $ban_exclude,
-				'ban_reason'		=> (string) $ban_reason,
-				'ban_give_reason'	=> (string) $ban_give_reason,
-			);
-		}
-
-		$db->sql_multi_insert(BANLIST_TABLE, $sql_ary);
-
-		// If we are banning we want to logout anyone matching the ban
-		if (!$ban_exclude)
-		{
-			switch ($mode)
-			{
-				case 'user':
-					$sql_where = 'WHERE ' . $db->sql_in_set('session_user_id', $banlist_ary);
-				break;
-
-				case 'ip':
-					$sql_where = 'WHERE ' . $db->sql_in_set('session_ip', $banlist_ary);
-				break;
-
-				case 'email':
-					$banlist_ary_sql = array();
-
-					foreach ($banlist_ary as $ban_entry)
-					{
-						$banlist_ary_sql[] = (string) str_replace('*', '%', $ban_entry);
-					}
-
-					$sql = 'SELECT user_id
-						FROM ' . USERS_TABLE . '
-						WHERE ' . $db->sql_in_set('user_email', $banlist_ary_sql);
-					$result = $db->sql_query($sql);
-
-					$sql_in = array();
-
-					if ($row = $db->sql_fetchrow($result))
-					{
-						do
-						{
-							$sql_in[] = $row['user_id'];
-						}
-						while ($row = $db->sql_fetchrow($result));
-
-						$sql_where = 'WHERE ' . $db->sql_in_set('session_user_id', $sql_in);
-					}
-					$db->sql_freeresult($result);
-				break;
-			}
-
-			if (isset($sql_where) && $sql_where)
-			{
-				$sql = 'DELETE FROM ' . SESSIONS_TABLE . "
-					$sql_where";
-				$db->sql_query($sql);
-
-				if ($mode == 'user')
-				{
-					$sql = 'DELETE FROM ' . SESSIONS_KEYS_TABLE . ' ' . ((in_array('*', $banlist_ary)) ? '' : 'WHERE ' . $db->sql_in_set('user_id', $banlist_ary));
-					$db->sql_query($sql);
-				}
-			}
-		}
-
-		// Update log
-		$log_entry = ($ban_exclude) ? 'LOG_BAN_EXCLUDE_' : 'LOG_BAN_';
-
-		// Add to admin log, moderator log and user notes
-		$phpbb_log->add('admin', $user->data['user_id'], $user->ip, $log_entry . strtoupper($mode), false, array($ban_reason, $ban_list_log));
-		$phpbb_log->add('mod', $user->data['user_id'], $user->ip, $log_entry . strtoupper($mode), false, array(
-			'forum_id' => 0,
-			'topic_id' => 0,
-			$ban_reason,
-			$ban_list_log
-		));
-		if ($mode == 'user')
-		{
-			foreach ($banlist_ary as $user_id)
-			{
-				$phpbb_log->add('user', $user->data['user_id'], $user->ip, $log_entry . strtoupper($mode), false, array(
-					'reportee_id' => $user_id,
-					$ban_reason,
-					$ban_list_log
-				));
-			}
-		}
-
-		$cache->destroy('sql', BANLIST_TABLE);
-
-		return true;
-	}
-
-	// There was nothing to ban/exclude. But destroying the cache because of the removal of stale bans.
-	$cache->destroy('sql', BANLIST_TABLE);
-
-	return false;
+	// TODO: logging
```

Comment:
Plan to implement within the current PR?

Reply:
Done inside the ban manager, will remove this comment